### PR TITLE
[SW-1718] Document client-less approach

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+The new approach is the base for supporting dynamic allocation, but already has several benefits:
+
+- Extended H2O Jars should no longer be used, but original H2O driver jars.
+- H2O client on Spark driver side can open the secured H2O cluster to the encapsulation environment
+  which might be security issue. With the new approach we don't use the H2O client and all
+  communication is using secured REST API connection, if configured.
+- No need to set client network or client ip address in case the driver and the H2O cluster
+  are running in different networks. We avoid connection issues related to these configurations and
+  simplify the deployment
+- Spark driver can now reconnect to a new or different H2O cluster. This can be done by running:
```
Comment (Tagar) [inline doc/src/site/sphinx/experimental/rest_api_based_client.rst]: that's awesome

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+This should reduce the complexity of deployment and brings more transparency as well.
+
+
+Benefits of the REST API based client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The new approach is the base for supporting dynamic allocation, but already has several benefits:
+
+- Extended H2O Jars should no longer be used, but original H2O driver jars.
+- H2O client on Spark driver side can open the secured H2O cluster to the encapsulation environment
```
Comment (Tagar) [inline doc/src/site/sphinx/experimental/rest_api_based_client.rst]: minor: didn't get phrase "encapsulation environment".

**Comment by Tagar** [comment]
thank you @jakubhava

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+The first iteration of replacing the H2O Node by REST API client is currently tested only
+under PySparkling in External Backend Mode. At this stage we do not support using PySparkling
+Algo API in this new approach. After the PySparkling Algo API is migrated to the new approach
+as well, we switch to the REST API Based client approach by default for PySparkling in external
+backend.
+
+Starting Sparkling Water with the REST API Client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable the REST API Based client, please start PySparkling with the spark option ``spark.ext.h2o.rest.api.based.client=true``.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/experimental/rest_api_based_client.rst]: nit: `REST API Based` -> `REST-API-based`?

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+This ensures that there won't be any H2O client running on the Spark driver side in both manual and automatic
+start of external H2O backend.
+
+Changes on the API
+~~~~~~~~~~~~~~~~~~
+The API remains unaffected for the users of automatic mode of external backend.
+
+For the users of the manual backend we have simplified the configuration and there is no need to specify cluster size anymore in advance.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/experimental/rest_api_based_client.rst]: nit: `a cluster size`?

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+Sparkling Water automatically discovers the cluster size. In particular ``spark.ext.h2o.external.cluster.size`` is not mandatory
+anymore for the users.
+
+The rest of the API also remains unaffected, except that the PySparkling Algo API is not currently supported in the
+first roll-out, but will follow up as soon as possible.
+
+Extended H2O Jars
+~~~~~~~~~~~~~~~~~
+We removed the requirement of using extended H2O jars. So in both manual and automatic mode of external backend you can
```
Comment (mn-mikke) [inline doc/src/site/sphinx/experimental/rest_api_based_client.rst]: nit: `the external backend`?

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+download the corresponding H2O jars for extended backend via the script ``./bin/get-h2o-driver.sh``. This script
```
Comment (mn-mikke) [inline doc/src/site/sphinx/experimental/rest_api_based_client.rst]: `extended backend` -> `the external backend`?

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/experimental/rest_api_based_client.rst]: Yup, like it!

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst (hunk 4 above)
Comment (jakubhava) [reply doc/src/site/sphinx/experimental/rest_api_based_client.rst]: 👍

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst (hunk 5 above)
Comment (jakubhava) [reply doc/src/site/sphinx/experimental/rest_api_based_client.rst]: 👍

File: doc/src/site/sphinx/experimental/rest_api_based_client.rst (hunk 6 above)
Comment (jakubhava) [reply doc/src/site/sphinx/experimental/rest_api_based_client.rst]: 👍

**Pull request merged**